nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutSQL.java

```diff
@@ -787,6 +802,29 @@ private void setParameter(final PreparedStatement stmt, final String attrName, f
                     stmt.setTimestamp(parameterIndex, new Timestamp(lTimestamp));
 
                     break;
+                case Types.BINARY:
+                case Types.VARBINARY:
+                case Types.LONGVARBINARY:
+                    byte[] bValue;
+
+                    switch(valueFormat){
+                        case "":
+                        case "ascii":
```

Comment:
How about cases for UTF-8 and UTF-16? Since the case just calls getBytes() with a charset name, we could support any/all of the ones in [StandardCharsets](https://docs.oracle.com/javase/8/docs/api/index.html?java/nio/charset/StandardCharsets.html). I found this while putting smileys in sql.args.N.value, the default handling will treat every byte as a 63.